pkg/controller/registry/reconciler/grpc.go

```diff
@@ -181,63 +196,54 @@ func (c *GrpcRegistryReconciler) ensurePod(source grpcCatalogSourceDecorator, ov
 	return nil
 }
 
-// ensureUpdatePod checks that for the same catalog source version the same imageID is running
+// ensureUpdatePod checks that for the same catalog source version the same container imageID is running
 func (c *GrpcRegistryReconciler) ensureUpdatePod(source grpcCatalogSourceDecorator) error {
 	if !source.Poll() {
 		return nil
 	}
-	var updateFlag bool
+
 	currentLivePods := c.currentPods(source)
 	currentUpdatePods := c.currentUpdatePods(source)
 
-	for _, updatePod := range currentUpdatePods {
-		if updatePodByDigest(updatePod, currentLivePods) {
-			logrus.WithField("CatalogSource", source.GetName()).Info("detect image update for catalogsource pod")
-
-			updateFlag = true
-
-			// Update the update pod to promote it to serving pod
-			err := c.SSAClient.Apply(context.TODO(), updatePod, func(p *v1.Pod) error {
-				p.Labels[CatalogSourceLabelKey] = source.GetName()
-				p.Labels[CatalogSourceUpdateKey] = ""
-				return nil
-			})()
-
-			if err != nil {
-				return errors.Wrapf(err, "error updating catalog source pod: %s", source.Pod().GetName())
-			}
+	if source.Update() && len(currentUpdatePods) == 0 {
```

Comment:
Is this `len(currentUpdatePods)` check necessary? When it is time to upgrade, we should just cleanup all update pod(s) and create a new one.

Reply:
I think this is safer because it ensures only 1 update pod is ever created on the cluster at a time. If the polling interval is really small (say 5 seconds) without this check we would end up spinning up a lot of extra pods and deleting them. Since we don't enforce a minimum polling interval, I think this is good to keep in-place. 

We delete immediately after checking the podID instead, to ensure the extra 1 pod doesn't hang around. The logic moved around a bit from the initial implementation, but I think the code is more intuitive this way.  

`if time to update and no update pod -> create update pod`
`...wait for update pod...`
`check update pod and immediately delete it, or delete the outdated pod`